### DISCUSSION TITLE
Add custom context

### DISCUSSION THREAD

**Discussion by flashtheman** [accounts-js/accounts#1152] [Q&A] [answered]
I would like to add a custom context so I don't have add the same code in each of my resolver functions. Is there some way to do this?

```
    // Create the Apollo Server that takes a schema and configures internal stuff
    const server = new ApolloServer({
      schema,
      context: accountsGraphQL.context,
    });
```

I would like to use it for giving each api request a typeorm Entity Manager instance so I can wrap each request into an SQL transaction.

**Comment by pradel** [comment, marked as answer]
Do you mean something like this?

```
 const server = new ApolloServer({
      schema,
      context: {
		 ...accountsGraphQL.context,
         myTypeormContextObject,
      },
    });
```

**Reply by flashtheman** [reply]
Thanks for your answer!

I tried that but if myTypeormContextObject is set to {x: 'xxx'} and I output the context in one of my resolver functions, then it only returns:
```
{"x":"xxx","_extensionStack":{"extensions":[]}}
```

Do you know the reason? Not sure if I'm doing something wrong. When I change it back to only accountsGrqphQL.context everything works normally.

**Reply by pradel** [reply]
Ah yeah sorry, my mistake, you have to do like this:
```
const server = new ApolloServer({
  typeDefs,
  resolvers,
  context: (req) => ({
    ...await accountsGraphQL.context(req),
    myTypeormContextObject,
  }),
});
```

**Reply by flashtheman** [reply]
Wow, thanks a lot!!!!! That works! You helped me a lot with that! Have a good weekend!

**Reply by pradel** [reply]
Cheers!
